controllers/modelmesh/runtime.go

```diff
@@ -229,8 +231,13 @@ func (m *Deployment) addRuntimeToDeployment(deployment *appsv1.Deployment) error
 				Name:  "RUNTIME_VERSION",
 				Value: runtimeVersion,
 			},
-			{}, {}, // allocate larger array to avoid reallocation
-		}[:7]
+		}
+
+		// To avoid reallocation, count the maximum number of env vars there could be
+		numVarsDefault := len(defaultEnvVars) // count of default env vars
+		numVarsOptional := 2                  // count of envs that may be added from annotations below
+		builtInAdapterContainer.Env = make([]corev1.EnvVar, numVarsDefault, numVarsDefault+numVarsOptional+len(rt.Spec.BuiltInAdapter.Env))
+		copy(builtInAdapterContainer.Env, defaultEnvVars)
```

Comment:
Not sure it's worth the extra complexity at this point, especially since this now does involve a separate array allocation. Maybe just revert the changes above here? Feel free to also remove the existing larger-array alloc logic if you don't like it.